hydromt_fiat/workflows/buildings.py

```diff
@@ -0,0 +1,241 @@
+import logging
+import numpy as np
+import xarray as xr
+from hydromt import gis_utils, raster
+
+
+logger = logging.getLogger(__name__)
+
+
+def create_population_per_building_map(
+    da_bld,
+    da_pop,
+    ds_like=None,
+    logger=logger,
+):
+    """Create a population per built-up grid cell layer.
+
+    Parameters
+    ----------
+    da_bld: xarray.DataArray
+        xarray.DataArray containing a numerical indication whether a grid cell is part of the built-up area or not.
+    pop_fn: xarray.DataArray
+        xarray.DataArray containing the number of inhabitants per grid cell.
+    ds_like: xarray.DataArray or xarray.DataSet
+        xarray.DataArray or xarray.DataSet containing the reference grid(s) projection.
+
+    Returns
+    -------
+    ds_count : xarray.DataSet
+        xarray.DataSet containing .
+    """
+
+    # Correct the buildings and population maps.
+    da_bld = da_bld.where(da_bld <= 0, other=1)
+    da_pop = da_pop.where(da_pop != da_pop.raster.nodata, other=0)
+    da_bld.raster.set_nodata(nodata=0)
+    da_pop.raster.set_nodata(nodata=0)
+
+    # Get area and density grids.
+    da_like_area = get_area_grid(ds_like)
+    da_bld_density = get_density_grid(da_bld).rename("bld_density")
+    da_pop_density = get_density_grid(da_pop).rename("pop_density")
+    da_bld_density.raster.set_nodata(nodata=0)
+    da_pop_density.raster.set_nodata(nodata=0)
+
+    # Get (average) grid resolutions in meters.
+    da_bld_res = get_grid_resolution(da_bld)
+    da_pop_res = get_grid_resolution(da_pop)
+    ds_like_res = get_grid_resolution(ds_like)
+
+    # downscaling exposure maps
+    if da_bld_res > ds_like_res or da_pop_res > ds_like_res:
+        logger.debug("Downscaling exposure maps to hazard resolution.")
+        if da_pop_res > da_bld_res:
+            da_low_res = da_pop
+            da_high_res = da_bld
+        else:
+            da_low_res = da_bld
+            da_high_res = da_pop
+
+        # get area map
+        da_bld_area = get_area_grid(da_bld).rename("bld_area")
+
+        # Create an index grid that connects the population and buildings maps.
+        da_idx = da_low_res.raster.nearest_index(
+            dst_crs=da_high_res.raster.crs,
+            dst_transform=da_high_res.raster.transform,
+            dst_width=da_high_res.raster.width,
+            dst_height=da_high_res.raster.height,
+        ).rename("index")
+        x_dim, y_dim = da_high_res.raster.x_dim, da_high_res.raster.y_dim
+        da_idx[x_dim] = da_high_res.raster.xcoords
+        da_idx[y_dim] = da_high_res.raster.ycoords
+
+        # # Create a population per buildings density map.
+        # use pandas as xarray groupby sum is slow, see https://github.com/pydata/xarray/issues/4473
+        df_sum = (
```

Comment:
fix speed: pandas based seems faster compared to xarray, see issue https://github.com/pydata/xarray/issues/4473